src/RequestHandler/inprocess/inprocessapplication.cpp

```diff
@@ -217,68 +225,86 @@ IN_PROCESS_APPLICATION::SetStdOut(
         fLocked = TRUE;
         if (!m_fDoneStdRedirect)
         {
-            hr = UTILITY::ConvertPathToFullPath(
-                m_pConfig->QueryStdoutLogFile()->QueryStr(),
-                m_pConfig->QueryApplicationPhysicalPath()->QueryStr(),
-                &struPath);
-            if (FAILED(hr))
-            {
-                goto Finished;
-            }
-
-            hr = UTILITY::EnsureDirectoryPathExist(struPath.QueryStr());
-            if (FAILED(hr))
-            {
-                goto Finished;
-            }
-
-            GetSystemTime(&systemTime);
-            hr = m_struLogFilePath.SafeSnwprintf(L"%s_%d%02d%02d%02d%02d%02d_%d.log",
-                struPath.QueryStr(),
-                systemTime.wYear,
-                systemTime.wMonth,
-                systemTime.wDay,
-                systemTime.wHour,
-                systemTime.wMinute,
-                systemTime.wSecond,
-                GetCurrentProcessId());
-            if (FAILED(hr))
-            {
-                goto Finished;
-            }
-
             saAttr.nLength = sizeof(SECURITY_ATTRIBUTES);
             saAttr.bInheritHandle = TRUE;
             saAttr.lpSecurityDescriptor = NULL;
-
-            m_hLogFileHandle = CreateFileW(m_struLogFilePath.QueryStr(),
-                FILE_WRITE_DATA,
-                FILE_SHARE_READ,
-                &saAttr,
-                CREATE_ALWAYS,
-                FILE_ATTRIBUTE_NORMAL,
-                NULL);
-
-            if (m_hLogFileHandle == INVALID_HANDLE_VALUE)
+            //
+            // CreatePipe for outputting stderr
+            if (!CreatePipe(&hStdErrReadPipe, &hStdErrWritePipe, &saAttr, 0 /*nSize*/))
             {
                 hr = HRESULT_FROM_WIN32(GetLastError());
                 goto Finished;
             }
+            // Verify it was inherited.
+            if (!SetHandleInformation(hStdErrReadPipe, HANDLE_FLAG_INHERIT, 0))
```

Comment:
what's the purpose the this sethandleinformation?

Reply:
This example: https://msdn.microsoft.com/en-us/library/windows/desktop/ms682499(v=vs.85).aspx used it for ensuring the read handle was not inherited.